docs-src/docs/guides/walkthrough.md

```diff
@@ -0,0 +1,317 @@
+# An Iterative Walkthrough
+
+This guide aims to take you through three increasingly-complex deployment examples using Layer0. Successive sections build upon the previous ones, and each deployment can be completed either through using the Layer0 CLI directly, or through Terraform using our custom Layer0 Terraform Provider.
+
+We assume that you're using Layer0 v0.9.0 or later. If you have not already installed and configured Layer0, see the [installation guide](/setup/install). If you are running an older version of Layer0, you may need to [upgrade](/setup/upgrade#upgrading-older-versions-of-layer0).
+
+If you intend to deploy services using the Layer0 Terraform Provider, you'll want to make sure that you've [installed](/reference/terraform-plugin/#install) the provider correctly.
+
+**Table of Contents**:
+
+- [Deployment 1](#deployment-1-a-simple-guestbook-app): A Simple Guestbook App
```

Comment:
[nit] Perhaps in titles/sections, we talk about the high-level process the walkthrough does instead of the particular services, e.g.:
```
[Part 1](#): Deploying a web service (Guestbook)
[Part 2](#): Deploying a web (Guestbook) and data store (Redis) service
[Part 3](#): Deploying a web (Guestbook), data store (Redis), and service discovery (Consul) service
```